lib/mutant/expression/namespace.rb

```diff
@@ -17,7 +17,7 @@ class Recursive < self
         def initialize(*)
           super
 
-          @syntax = "#{scope_name}*"
+          @syntax = "#{scope_name}*".freeze # rubocop:disable Style/RedundantFreeze
```

Comment:
Also curious if you know something I don't about this--I thought this was automatically frozen even with interpolation?

```ruby
# frozen_string_literal: true

class Foo
  def initialize
    @test = "#{rand}-test"
    pp @test.frozen?
  end
end

Foo.new # outputs `true`
```

Reply:
https://twitter.com/_m_b_j_/status/1342635659547340800

TLDR: MRI decided that, as each dstr gets allocated always freezing makes no sense.

I disagree as freezing is also a correctness mechanism, not just a memory optimization.

Instead: They should have patched MRI such as `+"some-#{dstr}"` would create a mutable DSTR, without frozen intermediary.

Reply:
...oh. I don't like that at all. :(